alacritty/src/window_context.rs

```diff
@@ -487,10 +491,7 @@ impl WindowContext {
             self.mouse.hint_highlight_dirty = false;
         }
 
-        // Request a redraw.
-        //
-        // Even though redraw requests are squashed in winit, we try not to
-        // request more if we haven't received a new frame request yet.
+        // We draw from `RedrawRequested`, so no need to request it, since we clear dirty bit.
         if self.dirty && !self.occluded && !matches!(event, WinitEvent::RedrawRequested(_)) {
             self.display.window.request_redraw();
         }
```

Comment:
The comment says we don't need to request it, but the code requests it. There's also no dirty bit clearing going on here, instead the dirty bit is being checked.

To me the two seem to have nothing to do with each other.

Reply:
The comment try to explain why we have `!matches`.